src/lib/types.ts

```diff
@@ -1,5 +1,46 @@
+import type { Breadcrumbs } from "$lib/components/Breadcrumbs";
+
 export type User = {
   email: string;
   name: string;
   avatarURL: string;
 };
+
+export type MetadataMapItem = {
```

Comment:
I've had a think about this, and while it would be nice to have a single type definition for this, I think this might be too generic. It's also divorced from the Contentful schema, which could make it brittle.

Instead of trying to reuse the existing `PageMetadataMapItem` type in `src/lib/loadEventsNewsMetadataMap.ts`, we should focus on fixing whatever is wrong with the type definition created by the `gen-operations` script that's getting output to `src/lib/$queries.generated.d.ts`.

Reply:
I think the problem that I was trying to solve here was that in order to reuse some code, the two types had some differences that did not overlap, so I wanted to create a type that was more flexible to support metadata map item of page, event, and news.

Reply:
I'll look into fixing this, but I may need some help with type generation.

Reply:
Posted this in Slack, but I think #583 should resolve some of the problems here (and correct the query).